src/packet.cpp

```diff
@@ -40,10 +39,10 @@ Packet::Packet(const std::array<uint8_t, 4096>& m_recvBuffer, size_t bytes, cons
     bool messageAuthenticator = false;
     for (size_t i = 0; i < m_attributes.size(); ++i)
     {
-        if (m_attributes[i]->type() == EAP_MESSAGE)
+        if (m_attributes[i]->type() == dict.attributes().code("EAP-Message"))
```

Comment:
Same as above, we can keed checking with EAP_MESSAGE (and MESSAGE_AUTHENTICATOR below).